pkg/metrics/smart/metrics.go

```diff
@@ -110,7 +175,99 @@ func (m *Metrics) WithBlockDeviceCurrentTemperatureValid() *Metrics {
 			Name:      "block_device_current_temperature_valid",
 			Help:      `Validity of the current temperature data reported. 0 means not valid, 1 means valid`,
 		},
-		[]string{"blockdevicename", "path", "hostname", "nodename"},
+		labels,
+	)
+	return m
+}
+
+// WithBlockDeviceRotationalLatency declares the metric rotational latency
+// as a prometheus metric
+func (m *Metrics) WithBlockDeviceRotationalLatency() *Metrics {
+	m.blockDeviceRotationalLatency = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: m.CollectorType,
+			Name:      "block_device_rotational_latency",
+			Help:      `Value of the latency in seconds. 0 indicates it is a SSD, non-zero value indicates a HDD`,
+		},
+		labels,
+	)
+	return m
+}
+
+// WithBlockDeviceCapacity declares the blockdevice capacity
+func (m *Metrics) WithBlockDeviceCapacity() *Metrics {
+	m.blockDeviceCapacity = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: m.CollectorType,
+			Name:      "block_device_capacity",
+			Help:      `Capacity of the block device in bytes`,
+		},
+		labels,
+	)
+	return m
+}
+
+// WithBlockDeviceRotationRate declares the rotational rate of a block device
+func (m *Metrics) WithBlockDeviceRotationRate() *Metrics {
+	m.blockDeviceRotationRate = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: m.CollectorType,
+			Name:      "block_device_rotational_rate",
+			Help:      `Rotational rate of the block device in RPM`,
+		},
+		labels,
+	)
+	return m
+}
+
+// WithBlockDeviceTotalBytesRead declares the total number of bytes read by a block device
+func (m *Metrics) WithBlockDeviceTotalBytesRead() *Metrics {
+	m.blockDeviceTotalReadBytes = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: m.CollectorType,
+			Name:      "block_device_total_read_bytes",
+			Help:      `total number of bytes read by a block device in bytes `,
+		},
+		labels,
+	)
+	return m
+}
+
+// WithBlockDeviceTotalBytesWritten declares the total number of bytes written by a block device
+func (m *Metrics) WithBlockDeviceTotalBytesWritten() *Metrics {
+	m.blockDeviceTotalWrittenBytes = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: m.CollectorType,
+			Name:      "block_device_total_written_bytes",
+			Help:      `total number of bytes written by a block device in bytes `,
+		},
+		labels,
+	)
+	return m
+}
+
+// WithBlockDeviceUtilizationRate declares the utilization rate of a block device
+func (m *Metrics) WithBlockDeviceUtilizationRate() *Metrics {
+	m.blockDeviceUtilizationRate = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: m.CollectorType,
+			Name:      "block_device_utilization_rate",
+			Help:      `utilization rate of a block device `,
```

Comment:
This is the utilization rate as a ratio of the vendor's expected workload wear factors. Values greater than 100% indicate the device has exceeded its designed workload. Can we change the help to be something like "ratio of actual workload vs manufacturer's designed workload for the device" ?